pages/workspaces/components/WorkspaceHeading/index.tsx

```diff
@@ -0,0 +1,79 @@
+import { useUpdateWorkspaceMutation } from '@/graphql/workspace/UpdateWorkspace.generated';
+import { Input, useColorModeValue } from '@chakra-ui/react';
+import { useEffect, useState } from 'react';
+
+const WorkspaceHeading: React.FC<{
+  workspaceName?: string;
+  workspaceId: string;
+  loading: boolean;
+}> = ({ workspaceName, workspaceId }) => {
+  const background = useColorModeValue('gray.200', 'gray.600');
+  const [isEditingName, setIsEditingName] = useState(false);
+  const [newName, setNewName] = useState(workspaceName ? workspaceName : '');
+  const [afterUpdateName, setAfterUpdateName] = useState('');
+  const [mutate] = useUpdateWorkspaceMutation();
+
+  const makeUpdate = async () => {
+    await mutate({ variables: { id: workspaceId, name: newName } });
+  };
+
+  // this is so the input element starts with the existing workspace name
+  // (which isn't immediately available)
+  useEffect(() => {
+    if (!workspaceName) {
+      return;
+    }
+
+    setNewName(workspaceName);
+  }, [workspaceName]);
+
+  // this is to clear the afterUpdateName once the workspaceName has been refetched
+  // and to clear the updating loading state
+  useEffect(() => {
+    if (workspaceName && afterUpdateName && workspaceName === afterUpdateName) {
+      setAfterUpdateName('');
+    }
+  }, [workspaceName, afterUpdateName]);
+
+  const saveUpdate = (el: HTMLInputElement, enterPressed = false) => {
```

Comment:
Just realized you could have saved yourself a param if you created a used a `ref` on the input 😄